### PR TITLE
Store Todos

### PR DESCRIPTION
**Pull request by steets250**
- [x] Update "Finish Pickup Event" explanation
- [x] Display order status on fulfillment page
- [x] Update order on fulfillment
- [x] Allow delete pickup events
- [x] Display orders in reverse cron order
- [x] Add stock check page
- [x] "You can buy 0 of this item"
- [x] Hide cancelled orders from pickup page

### CODE REVIEW

File: src/store/components/AdminFulfillPage/index.tsx [hunk 1]
```diff
@@ -122,6 +133,18 @@ const AdminFulfillPage: React.FC<AdminFulfillPageProps> = (props) => {
...
+                const aName = `${a.user.firstName} ${a.user.lastName}`;
+                const bName = `${b.user.firstName} ${b.user.lastName}`;
+
+                if (aName < bName) {
+                  return -1;
+                }
+                if (aName > bName) {
+                  return 1;
+                }
+                return 0;
```
Comment (yesean) [inline src/store/components/AdminFulfillPage/index.tsx:146]: why not use `String.prototype.localeCompare`?

File: src/store/components/AdminPickupPage/index.tsx [hunk 2]
```diff
@@ -165,6 +166,26 @@ const AdminPickupPage: React.FC<AdminPickupPageProps> = (props) => {
...
+                      disabled={submitting}
+                      onClick={() => {
+                        deletePickupEvent(pickupEvent?.uuid)
+                          .then(() => {
+                            notify('Success!', `Deleted ${pickupEvent?.title}.`);
+                            history.push('/store/admin');
+                          })
+                          .catch((reason) => {
+                            notify('API Error', reason.message || reason);
+                          });
```
Comment (yesean) [inline src/store/components/AdminPickupPage/index.tsx:184]: Is it common in the repo to inline multiline functions as a prop? I find these can make the JSX markup hard to read sometimes.

File: src/store/components/AdminPreparePage/index.tsx [hunk 3]
```diff
@@ -46,17 +46,19 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
-        name: item.option.item.itemName,
-        variantType: item.option.metadata?.type,
-        variantValue: item.option.metadata?.value,
-      };
-      merchData[`${item.option.uuid}`].quantity += 1;
+  pickupEvent.orders
+    ?.filter((order) => order.status !== OrderStatus.CANCELLED)
+    .forEach((order) => {
+      order.items.forEach((item) => {
+        merchData[`${item.option.uuid}`] = merchData[`${item.option.uuid}`] ?? {
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:53]: Does this need a template string? I believe `merchData[item.option.uuid]` would be the same

File: src/store/components/AdminPreparePage/index.tsx [hunk 4]
```diff
@@ -46,17 +46,19 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
+          quantity: 0,
+          name: item.option.item.itemName,
+          variantType: item.option.metadata?.type,
+          variantValue: item.option.metadata?.value,
+        };
+        merchData[`${item.option.uuid}`].quantity += 1;
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:59]: same as above

File: src/store/components/AdminPreparePage/index.tsx [hunk 5]
```diff
@@ -73,16 +75,26 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
-          }))}
+          dataSource={Object.values(merchData)
+            .sort((a, b) => {
+              if (a.name < b.name) {
+                return -1;
+              }
+              if (a.name > b.name) {
+                return 1;
+              }
+              return 0;
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:86]: same as above

File: src/store/components/AdminPreparePage/index.tsx [hunk 6]
```diff
@@ -73,16 +75,26 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
+            })
+            .map((data) => ({
+              uuid: data.name,
+              itemDisplay: (
+                <>
+                  <h2>{data.name}</h2>
+                  {data.variantType && data.variantValue && <h3>{`${data.variantType}: ${data.variantValue}`}</h3>}
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:93]: more of a nitpick and similar to the function prop but they be more readable if extracted into a function or separate boolean like `doesVariantExist`

File: src/store/components/AdminPreparePage/index.tsx [hunk 7]
```diff
@@ -102,38 +114,62 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
+              const nameA = `${a.user.firstName} ${a.user.lastName}`;
+              const nameB = `${b.user.firstName} ${b.user.lastName}`;
+
+              if (nameA < nameB) {
+                return -1;
+              }
+              if (nameA > nameB) {
+                return 1;
+              }
+              return 0;
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:129]: same as above

File: src/store/components/AdminPreparePage/index.tsx [hunk 8]
```diff
@@ -102,38 +114,62 @@ const AdminPreparePage: React.FC<AdminPreparePageProps> = (props) => {
...
+                  <ul>
+                    {updatedItems
+                      .sort((a, b) => {
+                        if (a.option.item.itemName < b.option.item.itemName) {
+                          return -1;
+                        }
+                        if (a.option.item.itemName > b.option.item.itemName) {
+                          return 1;
+                        }
+                        return 0;
```
Comment (yesean) [inline src/store/components/AdminPreparePage/index.tsx:161]: same as above

File: src/store/components/AdminQuantitiesPage/index.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+                    .sort((a, b) => {
+                      const textA = a.itemName.toUpperCase();
+                      const textB = b.itemName.toUpperCase();
+                      if (textA < textB) {
+                        return -1;
+                      }
+                      if (textA > textB) {
+                        return 1;
+                      }
+                      return 0;
```
Comment (yesean) [inline src/store/components/AdminQuantitiesPage/index.tsx:37]: same as above

File: src/store/components/AdminQuantitiesPage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+                            {item.options.length !== 1 &&
+                              item.options
+                                .sort((a, b) => {
+                                  if ((a.metadata?.position ?? 0) < (b.metadata?.position ?? 0)) {
+                                    return -1;
+                                  }
+                                  if ((a.metadata?.position ?? 0) > (b.metadata?.position ?? 0)) {
+                                    return 1;
+                                  }
+                                  return 0;
```
Comment (yesean) [inline src/store/components/AdminQuantitiesPage/index.tsx:54]: not sure if `position` is a number but I think `a.position - b.position` would suffice?

**Review by yesean — CHANGES_REQUESTED** [review]
Overall the PR looks pretty good! I left some comments about some stylistic things

File: src/store/components/AdminFulfillPage/index.tsx (hunk 1 above)
Comment (steets250) [reply src/store/components/AdminFulfillPage/index.tsx:146]: Good point, I swapped all string comparison instances with `localeCompare`. For some reason when I searched for JS string comparison this didn't pop up 🧐

File: src/store/components/AdminPickupPage/index.tsx (hunk 2 above)
Comment (steets250) [reply src/store/components/AdminPickupPage/index.tsx:184]: Good point; I pulled this out into a separate `handleDelete` function.

File: src/store/components/AdminPreparePage/index.tsx (hunk 3 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:53]: 👍

File: src/store/components/AdminPreparePage/index.tsx (hunk 4 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:59]: 👍

File: src/store/components/AdminPreparePage/index.tsx (hunk 5 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:86]: 👍

File: src/store/components/AdminPreparePage/index.tsx (hunk 6 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:93]: I pulled `data.variantType && data.variantValue` into a separate variable.

File: src/store/components/AdminPreparePage/index.tsx (hunk 7 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:129]: 👍

File: src/store/components/AdminPreparePage/index.tsx (hunk 8 above)
Comment (steets250) [reply src/store/components/AdminPreparePage/index.tsx:161]: 👍

File: src/store/components/AdminQuantitiesPage/index.tsx (hunk 9 above)
Comment (steets250) [reply src/store/components/AdminQuantitiesPage/index.tsx:37]: 👍

File: src/store/components/AdminQuantitiesPage/index.tsx (hunk 10 above)
Comment (steets250) [reply src/store/components/AdminQuantitiesPage/index.tsx:54]: Yep, I guess I didn't understand `.sort()` well enough before implementing

**Review by yesean — APPROVED** [review]
Everything looks good to me!

**Pull request merged**